File: frontend/src/api/buyer/consultation.ts
/**
 * 买家端咨询API
 */

import request from '@/utils/request'

// 咨询接口
export interface ConsultationDTO {
  productId: number
  contactName: string
  contactPhone?: string
  contactEmail?: string
  consultationContent: string
}

export interface ConsultationVO {
  id: number
  productId: number
  productName: string
  productImage: string
  userId?: number
  userName?: string
  contactName: string
  contactPhone?: string
  contactEmail?: string
  consultationContent: string
  replyContent?: string
  status: number
  statusText: string
  replyTime?: string
  replyAdminId?: number
  replyAdminName?: string
  createTime: string
  updateTime: string
}

/**
 * 提交咨询
 */
export const submitConsultation = (data: ConsultationDTO): Promise<void> => {
  return request.post('/api/buyer/consultation', data)
}

/**
 * 获取我的咨询列表
 */
export const getMyConsultations = (
  current: number,
  size: number
): Promise<{ records: ConsultationVO[]; total: number }> => {
  return request.get('/api/buyer/consultation/my', {
    params: { current, size }
  })
}

/**
 * 获取咨询详情
 */
export const getConsultationById = (consultationId: number): Promise<ConsultationVO> => {
  return request.get(`/api/buyer/consultation/${consultationId}`)
}

File: backend/src/main/java/com/shoppingmall/entity/Consultation.java
package com.shoppingmall.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 购买咨询实体
 */
@Data
@TableName("consultation")
public class Consultation {
    /**
     * 主键ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 商品ID
     */
    private Long productId;

    /**
     * 用户ID（可为空，支持匿名咨询）
     */
    private Long userId;

    /**
     * 联系人姓名
     */
    private String contactName;

    /**
     * 联系电话
     */
    private String contactPhone;

    /**
     * 联系邮箱
     */
    private String contactEmail;

    /**
     * 咨询内容
     */
    private String consultationContent;

    /**
     * 回复内容
     */
    private String replyContent;

    /**
     * 状态：0-待回复，1-已回复，2-已关闭
     */
    private Integer status;

    /**
     * 回复时间
     */
    private LocalDateTime replyTime;

    /**
     * 回复管理员ID
     */
    private Long replyAdminId;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}